src/cmd-build

```diff
@@ -429,3 +371,14 @@ else
     "${dn}"/prune_builds --workdir "${workdir}"
 fi
 rm builds/.build-commit
+
+# this is a bit of a hack: if qemu is in the target list, build it first. see
+# also start of script for comment about this.
+if [ -n "${targets[qemu]}" ]; then
+    "/usr/lib/coreos-assembler/cmd-buildextend-qemu"
```

Comment:
Won't this break if e.g. `ref_is_temp`?  In general, probably better to do the cleanup/pruning after this?

Reply:
It doesn't change the behaviour wrt. `ref_is_temp` because `buildextend-metal` also had the same semantics (and so `buildextend-qemu` also does now). Though testing it out, it turns out our behaviour was broken there. I pushed another commit to fix that.

>  In general, probably better to do the cleanup/pruning after this?

Hmm, I'd rather we keep it to the very end, since IMO we want e.g. `cosa build openstack` to be exactly equivalent to `cosa build ostree && cosa buildextend-qemu && cosa buildextend-openstack`.

This does highlight a behavioural change though, which is that `cosa build` can work on the OSTree part, but fail on the QEMU part, and we'd still end up with a new build. I think this is correct though, since it puts it on par with the other `buildextend-*` commands which work on an existing build.